lib/ansible/modules/systemd.py

```diff
@@ -503,11 +509,14 @@ def main():
                 elif module.params['state'] == 'stopped':
                     if is_running_service(result['status']) or is_deactivating_service(result['status']):
                         action = 'stop'
+                elif module.params['state'] == 'try-restarted':
+                    if is_running_service(result['status']):
+                        action = module.params['state'][:-2]
                 else:
                     if not is_running_service(result['status']):
                         action = 'start'
                     else:
-                        action = module.params['state'][:-2]  # remove 'ed' from restarted/reloaded
+                        action = module.params['state'][:-2]  # remove 'ed' from restarted/reloaded/try-restarted
```

Comment:
I don't think this will be executed if `state` is `try-restarted`.

Reply:
You're right, that was before I changed the structure.

Reply:
Fixed in https://github.com/ansible/ansible/pull/71172/commits/ad1cce5fb073d8b1de90dbdc77ed6cfcbc7c6b9a.